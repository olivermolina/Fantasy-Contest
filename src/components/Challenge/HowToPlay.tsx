import React from 'react';
import { Dialog, IconButton } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import Image from 'next/image';
import Link from 'next/link';

interface Props {
  handleClose: () => void;
  open: boolean;
}

const styles = {
  item: 'flex flex-col gap-2 bg-[#1A487F] p-4 rounded-lg',
};

export function HowToPlayDialog(props: Props) {
  return (
    <Dialog
      open={props.open}
      fullWidth
      maxWidth={'sm'}
      sx={(theme) => ({
        '& .MuiDialog-container': {
          '& .MuiPaper-root': {
            borderRadius: 5,
            backgroundColor: theme.palette.primary.main,
          },
        },
      })}
    >
      <HowToPlay {...props} />
    </Dialog>
  );
}

export default function HowToPlay(props: Props) {
  return (
    <div className={'flex flex-col gap-4 text-white bg-[#003370] p-4'}>
      <div className={'flex justify-between items-center'}>
        <span className={'text-xl font-semibold'}>How to Play</span>
        <IconButton onClick={props.handleClose}>
          <CloseIcon
            sx={(theme) => ({
              color: 'white',
              fontSize: 30,
              [theme.breakpoints.up('md')]: {
                fontSize: 40,
              },
            })}
          />
        </IconButton>
      </div>
      <div className={styles.item}>
        <div className={'flex gap-2 items-center text-sm'}>
          <span className={'rounded-full p-1 bg-[#003370] text-xs'}>01</span>
          <span>Choose 2 or more players from any sport</span>
        </div>
        <div className={'flex gap-2'}>
          <Image
            src="/assets/images/how-to-play-1.png"
            height={71}
            width={201}
            alt="Logo"
          />
        </div>
      </div>
      <div className={styles.item}>
        <div className={'flex gap-2 items-center text-sm'}>
          <span className={'rounded-full p-1 bg-[#003370] text-xs'}>02</span>
          <span>Pick MORE or LESS on their projected stats</span>
        </div>
        <div className={'flex gap-2'}>
          <Image
            src="/assets/images/how-to-play-2.png"
            height={75}
            width={160}
            alt="Logo"
          />
        </div>
      </div>
      <div className={styles.item}>
        <div className={'flex gap-2 items-center text-sm'}>
          <span className={'rounded-full p-1 bg-[#003370] text-xs'}>02</span>
          <span>Choose INSURED or ALL IN</span>
        </div>
        <div className={'flex gap-2'}>
          <Image
            src="/assets/images/how-to-play-3.png"
            height={71}
            width={265}
            alt="Logo"
          />
        </div>
      </div>
      <div className={'flex flex-row-reverse'}>
        <Link href="/contest-rules" legacyBehavior>
          <a
            className={'underline cursor-pointer'}
            target="_blank"
            rel="noopener noreferrer"
          >
            View Rules
          </a>
        </Link>
      </div>
    </div>
  );
}
