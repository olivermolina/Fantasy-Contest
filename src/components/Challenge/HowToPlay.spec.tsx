import React from 'react';
import { render } from '@testing-library/react';
import HowToPlay from './HowToPlay';

describe('HowToPlay', () => {
  it('should render correctly', () => {
    const props = {
      handleClose: () => console.log('handleClose'),
      open: false,
    };

    const { container } = render(<HowToPlay {...props} />);
    expect(container).toMatchSnapshot();
  });
});
