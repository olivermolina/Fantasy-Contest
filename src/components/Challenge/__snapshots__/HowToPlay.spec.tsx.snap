// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToPlay should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-4 text-white bg-[#003370] p-4"
  >
    <div
      class="flex justify-between items-center"
    >
      <span
        class="text-xl font-semibold"
      >
        How to Play
      </span>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1peaglv-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="flex flex-col gap-2 bg-[#1A487F] p-4 rounded-lg"
    >
      <div
        class="flex gap-2 items-center text-sm"
      >
        <span
          class="rounded-full p-1 bg-[#003370] text-xs"
        >
          01
        </span>
        <span>
          Choose 2 or more players from any sport
        </span>
      </div>
      <div
        class="flex gap-2"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27201%27%20height=%2771%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="flex flex-col gap-2 bg-[#1A487F] p-4 rounded-lg"
    >
      <div
        class="flex gap-2 items-center text-sm"
      >
        <span
          class="rounded-full p-1 bg-[#003370] text-xs"
        >
          02
        </span>
        <span>
          Pick MORE or LESS on their projected stats
        </span>
      </div>
      <div
        class="flex gap-2"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%2775%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="flex flex-col gap-2 bg-[#1A487F] p-4 rounded-lg"
    >
      <div
        class="flex gap-2 items-center text-sm"
      >
        <span
          class="rounded-full p-1 bg-[#003370] text-xs"
        >
          02
        </span>
        <span>
          Choose INSURED or ALL IN
        </span>
      </div>
      <div
        class="flex gap-2"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27265%27%20height=%2771%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="flex flex-row-reverse"
    >
      <a
        class="underline cursor-pointer"
        href="/contest-rules"
        rel="noopener noreferrer"
        target="_blank"
      >
        View Rules
      </a>
    </div>
  </div>
</div>
`;
